### PR TITLE
Cypress test for dashboard, app and folder CRUD operations

### PR DESCRIPTION
**Pull request by ajith-k-v**
Fixes: #3531
Screenshot 2022-07-26 at 7 22 27 PM

### CODE REVIEW

File: cypress-tests/cypress/e2e/dashboard/dashboard.cy.js [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    cy.get(dashboardSelector.userMenu).trigger("mouseover");
+    cy.get(dashboardSelector.profileLink).should(
+      "have.text",
+      dashboardText.profileLink
+    );
+    cy.get(dashboardSelector.logoutLink).should(
+      "have.text",
+      dashboardText.logoutLink
+    );
+    logout();
```
Comment (emidhun) [inline cypress-tests/cypress/e2e/dashboard/dashboard.cy.js:108]: Please let me know why we are using this logout.

File: cypress-tests/cypress/e2e/dashboard/dashboard.cy.js [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+      .and("have.text", commonText.modalYesButton);
+    cy.get(commonSelectors.modalCancelButton).click();
+    cy.get(commonSelectors.modalComponent).should("not.exist");
+
+    cy.get(commonSelectors.appCardOptions).first().click();
+    cy.get(commonSelectors.deleteAppOption).click();
+    cy.get(commonSelectors.modalYesButton).click();
+
+    cy.get(commonSelectors.appCard).contains(cloneAppName).should("not.exist");
+    cy.wait(1000);
```
Comment (emidhun) [inline cypress-tests/cypress/e2e/dashboard/dashboard.cy.js:257]: I think we can remove this hardcoded wait.

**Review by emidhun — CHANGES_REQUESTED** [review]
@ajith-k-v Please look at the comments and add modifications according to our last day's discussion.
Please let me know if you need any assistance from my side.
Thank you.

**Review by emidhun — COMMENTED** [review]
@ajith-k-v Can we add a custom assertion/method to validate visibility with any other attribute?

**Comment by ajith-k-v** [comment]
@emidhun Will work on that

**Review by emidhun — APPROVED** [review]
Screenshot 2022-08-04 at 4 57 50 PM
@ajith-k-v Looking good to me.
Thank you.

**Review by sherfin94 — APPROVED** [review]

**Pull request merged**